internal/server/boltdbstate/project.go

```diff
@@ -69,10 +69,185 @@ func (s *State) ProjectGet(ref *pb.Ref_Project) (*pb.Project, error) {
+			appRef := &pb.Ref_Application{
+				Application: app.Name,
+				Project:     project.Name,
+			}
+			if buildList, err := s.BuildList(appRef); err != nil {
+				return err
+			} else {
+				builds = append(builds, buildList...)
+			}
+			if artifactList, err := s.ArtifactList(appRef); err != nil {
+				return err
+			} else {
+				artifacts = append(artifacts, artifactList...)
+			}
+			if deploymentList, err := s.DeploymentList(appRef); err != nil {
+				return err
+			} else {
+				deployments = append(deployments, deploymentList...)
+			}
+			if releaseList, err := s.ReleaseList(appRef); err != nil {
+				return err
+			} else {
+				releases = append(releases, releaseList...)
+			}
+			if statusReportList, err := s.StatusReportList(appRef); err != nil {
+				return err
+			} else {
+				statusReports = append(statusReports, statusReportList...)
+			}
+
+			// Get app-scoped config
+			if appConfigVars, err := s.ConfigGet(&pb.ConfigGetRequest{
+				Scope: &pb.ConfigGetRequest_Application{Application: &pb.Ref_Application{
+					Project:     project.Name,
+					Application: app.Name,
+				}},
+			}); err != nil {
+				return err
+			} else {
+				configVars = append(configVars, appConfigVars...)
+			}
+		}
+
+		// Get project-scoped config
+		if projectConfigVars, err := s.ConfigGet(&pb.ConfigGetRequest{
+			Scope: &pb.ConfigGetRequest_Project{Project: ref},
+		}); err != nil {
+			return err
+		} else {
+			configVars = append(configVars, projectConfigVars...)
+		}
+
+		if workspaceList, err := s.ProjectListWorkspaces(ref); err != nil {
+			return err
+		} else {
+			for _, workspace := range workspaceList {
+				// Get the triggers for a project in the workspace
+				if triggerList, err := s.TriggerList(workspace.Workspace, &pb.Ref_Project{Project: project.Name}, nil, []string{}); err != nil {
+					return err
+				} else {
+					triggers = append(triggers, triggerList...)
+				}
+				if workspaceDetail, err := s.WorkspaceGet(workspace.Workspace.Workspace); err != nil {
+					return err
+				} else {
+					// If the project we're deleting is the only project in the workspace, we delete the workspace
+					// We don't delete the default workspace
+					if len(workspaceDetail.Projects) == 1 && workspace.Workspace.Workspace != "default" {
+						workspaces = append(workspaces, workspace)
+					}
+				}
+
+			}
+		}
+
+		if pipelines, err = s.PipelineList(ref); err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	// Builds, artifacts, deployments, releases, status reports, pipelines, triggers,
+	// workspaces and config are deleted with a project
+	// Jobs and tasks will NOT be deleted along with a project
+	// Instances are expected to be deleted before ProjectDelete, via the destroy op
+	for _, build := range builds {
+		if err = s.BuildDelete(&pb.Ref_Operation{Target: &pb.Ref_Operation_Id{Id: build.Id}}); err != nil {
+			return err
+		}
+		s.log.Debug("deleted build " + build.Id)
```

Comment:
Should we emit a debug log for all the other things we delete? This is the only one

Reply:
I included this during testing the deletion of multiple builds for an app in a project. I think it's worth adding a debug log for other deleted things, so that we can see what was successfully deleted if desired.